src/variety/lither.js

```diff
@@ -0,0 +1,327 @@
+		},
+
+		UNDECIDED_NUM_XML: 5,
+		PBOX_ADJUST: 1,
+		decodeBorderLine_lither_XMLAnswer: function() {
+			this.decodeCellXMLArow(function(cross, name) {
+				var val = 0;
+				var bdh = cross.relbd(0, 1),
+					bdv = cross.relbd(1, 0);
+				if (name.charAt(0) === "n") {
+					val = +name.substr(1);
+				} else {
+					if (name.match(/h/)) {
+						val += 1;
+					}
+					if (name.match(/v/)) {
+						val += 2;
+					}
+				}
+				if (val & 1) {
+					bdh.line = 1;
+				}
+				if (val & 2) {
+					bdv.line = 1;
+				}
+				if (val & 4) {
+					bdh.qsub = 2;
+				}
+				if (val & 8) {
+					bdv.qsub = 2;
+				}
+			});
+		},
+		encodeBorderLine_lither_XMLAnswer: function() {
+			this.encodeCellXMLArow(function(cross) {
+				var val = 0,
+					nodename = "";
+				var bdh = cross.relbd(0, 1),
+					bdv = cross.relbd(1, 0);
+				if (bdh.line === 1) {
+					val += 1;
+				}
+				if (bdv.line === 1) {
+					val += 2;
+				}
+				if (bdh.qsub === 2) {
+					val += 4;
+				}
+				if (bdv.qsub === 2) {
+					val += 8;
+				}
+
+				if (val === 0) {
+					nodename = "s";
+				} else if (val === 1) {
+					nodename = "h";
+				} else if (val === 2) {
+					nodename = "v";
+				} else if (val === 3) {
+					nodename = "hv";
+				} else {
+					nodename = "n" + val;
+				}
+				return nodename;
+			});
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkLineExist+",
+			"checkPluralLine",
+			"checkAllBranchOrTerminate",
+			"checkdir4BorderLine",
+			"checkNoLoop"
+		],
+
+		checkdir4BorderLine: function() {
+			this.checkAllCell(function(cell) {
+				return cell.qnum >= 0 && cell.getdir4BorderLine1() !== cell.qnum;
+			}, "nmLineNe");
+		},
+
+		checkAllBranchOrTerminate: function() {
+			var result = true,
+				bd = this.board;
+			if (!bd.linegraph.ltotal[0] && !bd.linegraph.ltotal[2]) {
+				return;
+			}
+			var boardcross = bd.cross;
+			for (var c = 0; c < boardcross.length; c++) {
+				var cross = boardcross[c];
+				if (cross.lcnt !== 0 && cross.lcnt !== 2) {
+					continue;
+				}
+
+				result = false;
+				if (this.checkOnly) {
+					break;
+				}
+				cross.seterr(1);
+				bd.borderinside(
+					cross.bx - 1,
+					cross.by - 1,
+					cross.bx + 1,
+					cross.by + 1
+				).seterr(1);
+			}
+			if (!result) {
+				this.failcode.add("lnNoBranchOrTerm");
+			}
+		},
+
+		checkPluralLine: function() {
+			var bd = this.board,
+				paths = bd.linegraph.components;
+			if (paths.length < 2) {
+				this.failcode.add("lnSnLine");
+				bd.border.setnoerr();
+				if (paths[0]) {
+					paths[0].setedgeerr(1);
+				}
+			}
+		},
+
+		checkNoLoop: function() {
+			var bd = this.board,
+				paths = bd.linegraph.components;
+			for (var i = 0; i < paths.length; i++) {
+				if (paths[i].circuits === 0) {
+					continue;
+				}
+
+				this.failcode.add("lnHasLoop");
+				if (this.checkOnly) {
+					return;
+				}
+				paths[i].setedgeerr(1);
```

Comment:
When marking borders as an error, all other borders should be greyed out. This can be done by calling `bd.border.setnoerr();`.

Reply:
Fixed

Reply:
`setnoerr` should only happen if an error is actually found, not when starting the check.

Reply:
Fixed